.github/workflows/code-styling.yml

```
@@ -45,25 +73,25 @@ jobs:
             styler::style_file()
         shell: Rscript --no-init-file {0}
 
-
       ## Second, style the python code ##
-      - name: Format python files
+      - name: Style Python code for each analysis module
         uses: chartboost/ruff-action@v1
         with:
-          args: format analyses/
+          args: format
+          src: "analyses/${{ matrix.module }}/"
 
       - name: Create PR with styled code
         uses: peter-evans/create-pull-request@v6 # defaults to ${GITHUB_TOKEN} token
         with:
           commit-message: Styled code
           signoff: false
-          branch: style-analysis-modules
+          branch: style-module_${{ matrix.module }}
           base: main
           delete-branch: true
-          title: "GHA: Automated analysis module code styling"
+          title: "GHA: Automated code styling for analysis module: ${{ matrix.module }}"
           body: |
             ### Description:
-            This PR was auto-generated by GitHub Actions to style all R (scripts and notebooks) and Python (scripts only) code in `analyses`.
+            This PR was auto-generated by GitHub Actions to style all R (scripts and notebooks) and Python (scripts only) code in `analyses/${{ matrix.module }}`.
```

ruff does notebooks too as of version 0.6.0, and I am pretty sure this action gets the latest version by default. 
```suggestion
            This PR was auto-generated by GitHub Actions to style all R and Python code (in scripts and notebooks) in `analyses/${{ matrix.module }}`.
```

---

yup, default: https://github.com/ChartBoost/ruff-action?tab=readme-ov-file#advanced-usage